src/tests/tracing/eventpipe/allocationsampling/allocationsampling.cs

```diff
@@ -0,0 +1,255 @@
+            //GC.Collect();
+
+            // run the same allocations test with no sampling, AllocationTick and AllocationSampled
+            Stopwatch clock = new Stopwatch();
+            clock.Start();
+            int allocatedSize = AllocateObjects(InstanceCount * 50);
+            clock.Stop();
+            var duration = clock.ElapsedMilliseconds;
+            Logger.logger.Log($"       #GCs: {GC.CollectionCount(0)}");
+            Logger.logger.Log($"No Sampling: {InstanceCount * 50} instances allocated for {allocatedSize} bytes in {duration} ms");
+            _noSamplingDuration = duration;
+            Logger.logger.Log("-");
+
+            //// trigger GC to avoid one during the next tests
+            //GC.Collect();
+
+            var ret = IpcTraceTest.RunAndValidateEventCounts(
+                new Dictionary<string, ExpectedEventCount>() { { "Microsoft-Windows-DotNETRuntime", -1 } },
+                _eventGeneratingActionForAllocationsWithAllocationTick,
+                // GCKeyword (0x1): 0b1
+                new List<EventPipeProvider>() { new EventPipeProvider("Microsoft-Windows-DotNETRuntime", EventLevel.Verbose, 0b1) },
+                1024, _DoesTraceContainAllocationTickEvents, enableRundownProvider: false);
+            if (ret != 100)
+                return ret;
+            Logger.logger.Log("-");
+
+            //// trigger GC to avoid one during the next tests
+            //GC.Collect();
+
+            ret = IpcTraceTest.RunAndValidateEventCounts(
+                new Dictionary<string, ExpectedEventCount>() { { "Microsoft-Windows-DotNETRuntime", -1 } },
+                _eventGeneratingActionForAllocationsWithAllocationSampled,
+                // AllocationSamplingKeyword (0x80000000000): 0b1000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000
+                new List<EventPipeProvider>() { new EventPipeProvider("Microsoft-Windows-DotNETRuntime", EventLevel.Informational, 0x80000000000) },
+                1024, _DoesTraceContainAllocationSampledEvents, enableRundownProvider: false);
+            if (ret != 100)
+                return ret;
+
+            return 100;
+        }
+
+        private static int AllocateObjects(int instanceCount)
+        {
+            int size = 0;
+            for (int i = 0; i < instanceCount; i++)
+            {
+                byte[] bytes = new byte[35];
+                size += bytes.Length;
+            }
+
+            return size;
+        }
+
+        // 2000 instances of 1KB byte arrays should trigger around 20 events with the default 1/100KB sampling rate
+        const int InstanceCount = 2000;
+        const int MinExpectedEvents = 15;
+
+        // allocate objects to trigger dynamic allocation sampling events
+        private static Action _eventGeneratingActionForAllocations = () =>
+        {
+            int allocatedSize = 0;
+            for (int i = 0; i < InstanceCount; i++)
+            {
+                if ((i != 0) && (i % InstanceCount/5 == 0))
```

Comment:
```suggestion
                if ((i != 0) && (i % (InstanceCount/5) == 0))
```
% and / have the same precedence for order of operations so this logging probably wasn't doing what you wanted.

Reply:
Good catch  :^)
And the log was also wrong...